File: DFS.java
/*
Author: Akhilesh Borgaonkar
My attempt for DFS for BT
*/

public class DFS {

    static class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode(int data) { val = data; }
    }

    static void printInorder(TreeNode node) {
        if(node == null)
            return;

        printInorder(node.left);
        System.out.println(node.val);
        printInorder(node.right);
    }

    public static void main(String[] args) {

        TreeNode treeNode = new TreeNode(3);
        treeNode.left = new TreeNode(9);
        treeNode.left.left = new TreeNode(12);
        treeNode.right = new TreeNode(20);
        treeNode.right.left = new TreeNode(15);
        treeNode.right.right = new TreeNode(7);

        printInorder(treeNode);
    }
}

File: DFS.py
#Time Complexity: O(n)
#Space Complexity: O(h)
#Level order traversal using DFS

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    global output
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root is None:
            return
        self.output=[]
        self.DFS(root,0)
        return self.output

    def DFS(self, root, depth):
        if root is None:
            return
        if depth == len(self.output):
            self.output.append([])
        self.output[depth].append(root.val)
        self.DFS(root.left, depth+1)
        self.DFS(root.right, depth+1)

     



